TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/middleware/backfill.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from typing import Dict, Tuple
+from functools import partial
+
+from console_link.middleware.error_handler import handle_errors
+from console_link.middleware.json_support import support_json_return
+from console_link.models.backfill_base import Backfill, BackfillStatus
+from console_link.models.command_result import CommandResult
+from console_link.models.utils import ExitCode
+
+
+logger = logging.getLogger(__name__)
+
+
+@support_json_return()
+def describe(backfill: Backfill, as_json=False) -> Tuple[ExitCode, Dict]:
+    response = backfill.describe()
+    return (ExitCode.SUCCESS, response)
+
+
+handle_backfill_errors = partial(handle_errors, service_type="backfill")
+
+
+@handle_backfill_errors(on_success=lambda result: (ExitCode.SUCCESS,
+                                                   "Backfill created successfully." + "\n" + result))
+def create(backfill: Backfill, *args, **kwargs) -> CommandResult[str]:
+    logger.info(f"Creating backfill with {args=} and {kwargs=}")
+    return backfill.create(*args, **kwargs)
+
+
+@handle_errors("backfill",
+               on_success=lambda result: (ExitCode.SUCCESS, "Backfill started successfully." + "\n" + result))
+def start(backfill: Backfill, *args, **kwargs) -> CommandResult[str]:
+    logger.info("Starting backfill")
+    return backfill.start(*args, **kwargs)
+
+
+@handle_errors("backfill",
+               on_success=lambda result: (ExitCode.SUCCESS, "Backfill stopped successfully." + "\n" + result))
+def stop(backfill: Backfill, *args, **kwargs) -> CommandResult[str]:
+    logger.info("Stopping backfill")
+    return backfill.stop(*args, **kwargs)
+
+
+@handle_errors("backfill",
+               on_success=lambda status: (ExitCode.SUCCESS, f"{status[0]}\n{status[1]}"))
+def status(backfill: Backfill, deep_check: bool, *args, **kwargs) -> CommandResult[Tuple[BackfillStatus, str]]:
+    logger.info(f"Getting backfill status with {deep_check=}")
+    return backfill.get_status(deep_check, *args, **kwargs)
+
+
+@handle_errors("backfill",
+               on_success=lambda status: (ExitCode.SUCCESS, status))
+def scale(backfill: Backfill, units: int, *args, **kwargs) -> CommandResult[str]:
+    logger.info(f"Scaling backfill to {units} units")
+    return backfill.scale(units, *args, **kwargs)
```

Comment:
Can you help me understand what *args and **kwargs are intended for if we're passing all the variables directly like units here

Reply:
That's a good question. The way I've been thinking about is that it's for optional params that might apply to one type of service but not another, and this ensures they get passed all the way through. So far, we've almost entirely avoided actually having that kind of parameter, so putting these in place may have been premature.

Reply:
I think we definitely want to explicitly specify the params that are core to the functionality (like `units`), but I don't have a clear sense of whether we'll actually need the rest. So far, we've done well at pushing everything we can into the services.yaml.